### PR TITLE
Added _total suffix to OTEL counter metrics.

### PR DESCRIPTION
**Pull request by Wise-Wizard**
**Which problem is this PR solving?**

This PR addresses a part of the issue [#5633 ](https://github.com/jaegertracing/jaeger/issues/5633)

**Description of the changes**
This is a PR to achieve Observability Parity in metrics between V1 and V2 components by configuring OTEL Collector to emit desired metrics.
**How was this change tested?**

The changes were tested by running the following command:

```bash
make test
```
```bash
CI actions and compare.py script
```
**Checklist**

- [x] I have read [CONTRIBUTING_GUIDELINES.md](https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md)
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - `for jaeger: make lint test`
  - `for jaeger-ui: yarn lint` and `yarn test`

### CODE REVIEW

File: internal/metrics/otelmetrics/factory.go [hunk 1]
```diff
@@ -131,3 +131,10 @@ func attributeSetOption(tags map[string]string) metric.MeasurementOption {
 	}
 	return metric.WithAttributes(attributes...)
 }
+
+func counterNamingConvention(name string) string {
+	if !strings.HasSuffix(name, "_total") {
+		name += "_total"
```
Comment (yurishkuro) [inline internal/metrics/otelmetrics/factory.go:137]: did we have a precedent of doing this elsewhere in Jaeger?

**Review by yurishkuro — COMMENTED** [review]
please add a unit test

File: internal/metrics/otelmetrics/factory.go (hunk 1 above)
Comment (Wise-Wizard) [reply internal/metrics/otelmetrics/factory.go:137]: Yes, this was being done in prometheus metrics lib.

**Comment by Wise-Wizard** [comment]
> please add a unit test

Sorry, can you specify which Unit test? Because, the tests are already written for counter using _total suffix.

**Comment by yurishkuro** [comment]
> Sorry, can you specify which Unit test? Because, the tests are already written for counter using _total suffix.

you are adding a new behavior. If you remove that behavior, some test should fail.

**Comment by Wise-Wizard** [comment]
> > Sorry, can you specify which Unit test? Because, the tests are already written for counter using _total suffix.
> 
> you are adding a new behavior. If you remove that behavior, some test should fail.

No, that's the thing without adding that functionality itself the metric names are being added with _total suffix in the tests itself.
In the tests I had to compare with counter_total to pass them.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**